packages/catalog-client/src/types/api.ts

```diff
@@ -269,6 +270,16 @@ export type AddLocationResponse = {
   exists?: boolean;
 };
 
+/**
+ * The response type for {@link CatalogClient.validateEntity}
+ *
+ * @public
+ */
+export type ValidateEntityResponse = {
+  valid: boolean;
+  errors?: SerializedError[];
+};
```

Comment:
is it worth making this one `{ valid: true } | { valid: false; errors: SerializedError[] }`?

Reply:
Good idea. Done.